src/install.rs

```diff
@@ -67,32 +67,51 @@ pub fn install(config: &InstallConfig) -> Result<()> {
     ensure_exclusive_access(&config.device)
         .chain_err(|| format!("checking for exclusive access to {}", &config.device))?;
 
+
+
+    let mut extrapart = Disk::get_extra_gptpartitions(&config.device);
+    println!("Extra Partitions = {}\n",extrapart.len());
+
     // get reference to partition table
     // For kpartx partitioning, this will conditionally call kpartx -d
     // when dropped
     let mut table = Disk::new(&config.device)
         .get_partition_table()
         .chain_err(|| format!("getting partition table for {}", &config.device))?;
 
+    if config.wipedisk {
+       println!("Erasing partition table\n");
+       clear_partition_table(&mut dest, &mut *table)?;
+       extrapart.truncate(0);
+       }
+
     // copy and postprocess disk image
     // On failure, clear and reread the partition table to prevent the disk
     // from accidentally being used.
     if let Err(err) = write_disk(&config, &mut source, &mut dest, &mut *table) {
         // log the error so the details aren't dropped if we encounter
         // another error during cleanup
         eprint!("{}", ChainedError::display_chain(&err));
-
+ 
+        // Default is now to preserve on error - So flag is changed to allow a forced clear
         // clean up
-        if config.preserve_on_error {
-            eprintln!("Preserving partition table as requested");
-        } else {
+        if config.clear_on_error {
             clear_partition_table(&mut dest, &mut *table)?;
+        } else {
+            eprintln!("Preserving partition table as requested");
+            Disk::add_extra_gptpartitions(&config.device,extrapart,config.dsneeded)
+                  .expect("Failed to add back additional partitions");
         }
 
         // return a generic error so our exit status is right
         bail!("install failed");
     }
 
+    // Make sure end_lba and partition table is consistent 
+    Disk::update_gpt_headers(&config.device);
+
+    Disk::add_extra_gptpartitions(&config.device,extrapart,config.dsneeded)
```

Comment:
This update headers is because there is a mismatch in the headers that can cause gptman issues, and most partitioning software. This was added to force it to be clean before anything else. Gptman may fail to do changes otherwise.